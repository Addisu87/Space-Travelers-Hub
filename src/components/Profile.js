import React from 'react';
import { useSelector } from 'react-redux';
import './styles/Profile.css';

const Profile = () => {
  const rockets = useSelector((state) => state.Rockets);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved);
  return (
    <>
      <div className="profile-container">
        <div className="profile-missions">
          <h2>My Missions</h2>
        </div>
        <div className="profile-rockets">
          <h2>My Rockets</h2>
          {reservedRockets.length ? (
            <div className="reservation-container">
              {reservedRockets.map((rocket) => (
                <p id="reserved-rocket" key={rocket.id}>
                  {rocket.name}
                </p>
              ))}
            </div>
          ) : null}
        </div>
      </div>
    </>
  );
};

export default Profile;
