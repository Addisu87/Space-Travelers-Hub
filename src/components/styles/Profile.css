.profile-container {
  display: flex;
  justify-content: space-around;
}

.reservation-container {
  border: 1px solid #8c8c8e;
  border-radius: 10px;
  margin-top: 0.5rem;
}

#reserved-rocket {
  border-top: 1px solid #8c8c8e;
  padding: 1rem;
}

#reserved-rocket:nth-child(1) {
  border-top: none;
}
