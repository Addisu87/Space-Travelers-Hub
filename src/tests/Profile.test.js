import React from "react";
import renderer from "react-test-renderer";
import Profile from '../components/Profile';

describe("Jest Snapshot testing suite", () => {
  it("Matches DOM Snapshot", () => {
    const domTree = renderer.create(<Profile />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});