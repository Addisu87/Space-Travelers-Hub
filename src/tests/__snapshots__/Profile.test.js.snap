// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<div
  class="profile-container"
>
  <div
    class="profile-missions"
  >
    <h2>
      My Missions
    </h2>
  </div>
  <div
    class="profile-rockets"
  >
    <h2>
      My Rockets
    </h2>
  </div>
</div>
`;
